client/src/languages/rPython.ts

```diff
@@ -0,0 +1,327 @@
+		let previewButton = h('button', { onclick:() => context.evaluate(cell) }, ["Preview"])
+		let triggerSelect = (t:number) => context.trigger({kind:'switchtab', index: t})
+	let preview = h('div', {}, [(cell.code.value==undefined) ? previewButton : (printPreview(triggerSelect, state.tabID, <Values.DataFrame>cell.code.value))]);
+		let code = createEditor("python", state.block.source, cell, context)
+		return h('div', { }, [code, preview])
+	}
+}
+
+export class apiPlugin implements Langs.LanguagePlugin {
+	readonly language: string;
+	readonly editor: Langs.Editor<rPythonState, rPythonEvent>;
+	readonly serviceURI: string;
+
+	constructor(l: string, uri: string) {
+		this.language = l;
+		this.serviceURI = uri;
+		this.editor = rPythonEditor;
+	}
+
+	async evaluate(node:Graph.Node) : Promise<Values.Value> {
+		let pyNode = <Graph.PyNode>node
+	
+		async function getValue(blob) : Promise<Values.Value> {
+			var pathname = new URL(blob).pathname;
+			let headers = {'Content-Type': 'application/json'}
+			let url = DATASTORE_URI.concat(pathname)
+			try {
+				const response = await axios.get(url, {headers: headers});
+				return response.data
+			}
+			catch (error) {
+				console.error(error);
+			}
+		}
+	
+		async function getEval(body, serviceURI) : Promise<Values.ExportsValue> {
+			let url = serviceURI.concat("/eval")
+			let headers = {'Content-Type': 'application/json'}
+			try {
+				const response = await axios.post(url, body, {headers: headers});
+				var results : Values.ExportsValue = {}
+				for(var df of response.data.frames) 
+				// results[df.name] = await getValue(df.url)
+				results[df.name] = {url: df.url, data: await getValue(df.url)}
+				return results;
+			}
+			catch (error) {
+				console.error(error);
+			}
+		}
+	
+		switch(pyNode.kind) {
+			case 'code': 
+				let importedFrames : { name:string, url:string }[] = [];
+				for (var ant of pyNode.antecedents) {
+				let imported = <Graph.ExportNode>ant
+				// console.log(imported);
+				// console.log(imported.value.data);
+				importedFrames.push({ name: imported.variableName, url: (<Values.DataFrame>imported.value).url })
+				}
+						let src = pyNode.source
+						let hash = Md5.hashStr(src)
+						let body = {"code": src,
+										"hash": hash,
+										"frames": importedFrames}
+						return await getEval(body, this.serviceURI);
+			case 'export':
+				let pyExportNode = <Graph.PyExportNode>node
+				let exportNodeName= pyExportNode.variableName
+				let exportsValue = <Values.ExportsValue>pyExportNode.code.value
+				return exportsValue[exportNodeName]
+			}
+	}
+
+	parse (code:string) {	
+		return new rPythonBlockKind(code, this.language);
+	}
+
+	bind (scopeDictionary: Langs.ScopeDictionary, block: Langs.Block) {
+			let pyBlock = <rPythonBlockKind>block
+			// return getCodeExports(scopeDictionary, pyBlock.source)
+			let dependencies:Graph.PyExportNode[] = [];
+			let node:Graph.PyCodeNode = {
+				language:"python", 
```

Comment:
This should be the language provided when constructing the plugin - not `python`.